File: 프로그래머스/2/131127. 할인 행사/할인 행사.java
import java.util.*;
class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        int answer = 0;
        
        for(int i=0; i< discount.length-9; i++){ //100,000
            String[] sliced = Arrays.copyOfRange(discount, i, i+10);
            List<String> tmp = Arrays.asList(sliced);
            int count = 0;
            for(int j=0; j<want.length; j++){
                String now = want[j];
                int total = (int) tmp.stream().filter(o -> o.equals(now)).count();
                if(total == number[j]){
                    count += total;
                }else{
                    continue;
                }
            }
            
            if(count == 10){
                answer += 1;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0
    n = len(discount)
    m = len(want)
    
    for i in range(n):
        tmp = 0
        for j in range(m):
            if discount[i:i+10].count(want[j]) == number[j]:
                tmp += 1
        
        if tmp == m:
            answer += 1
            
    return answer